hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/AlertsService.java

```diff
@@ -218,6 +218,15 @@ void resolveAlertsForTrigger(String tenantId, String triggerId, String resolvedB
     void sendData(Collection<Data> data) throws Exception;
 
     /**
+     * Send data into the alerting system for evaluation.
+     *
+     * @param data Not Null.  The data to be evaluated by the alerting engine.
+     * @param ignoreFiltering  An optimization. Set true *only* if you are sure the data is useful for evaluation.
+     * @throws Exception any problem.
+     */
+    void sendData(Collection<Data> data, boolean ignoreFiltering) throws Exception;
```

Comment:
In general this flag would not be used.  It is only when a client is sure they don't need the incoming data to be filter-checked.  In this case metrics would be the only java client in the foreseeable future. A config prop is probably not appropriate because it is client-specific, perhaps even call-specific for a particular client.